setuptools/command/bdist_wheel.py

```diff
@@ -291,6 +288,21 @@ def finalize_options(self) -> None:
         if self.build_number is not None and not self.build_number[:1].isdigit():
             raise ValueError("Build tag (build-number) must start with a digit.")
 
+    def _validate_py_limited_api(self) -> None:
+        if not self.py_limited_api:
+            return
+
+        if not re.match(PY_LIMITED_API_PATTERN, self.py_limited_api):
+            raise ValueError(f"py-limited-api must match '{PY_LIMITED_API_PATTERN}'")
+
+        if sysconfig.get_config_var("Py_GIL_DISABLED"):
+            raise ValueError(
+                f"`py_limited_api={self.py_limited_api!r}` not supported. "
+                "`Py_LIMITED_API` is currently incompatible with "
+                f"`Py_GIL_DISABLED` ({sys.abiflags=!r}). "
+                "See https://github.com/python/cpython/issues/111506."
+            )
```

Comment:
@jaraco do you agree with this approach?

Reply:
Sounds good to me. I don't have more than a superficial understanding of the concern. For example, it's unclear to me if this limitation is temporary and if so, under what conditions could the restriction be lifted. I hope in the future some people will bring more insight and help provide more robust support for the various combinations (if feasible).